airflow/api_fastapi/core_api/routes/public/dag_run.py

```diff
@@ -362,16 +362,19 @@ def trigger_dag_run(
                 end=pendulum.instance(body.data_interval_end),
             )
         else:
-            data_interval = dag.timetable.infer_manual_data_interval(run_after=logical_date)
+            data_interval = dag.timetable.infer_manual_data_interval(
+                run_after=logical_date or pendulum.now("UTC")
```

Comment:
I think we should have only one `pendulum.now("UTC")` so it's easy to identify that all of those were automatically filled with `now`. Here we will have multiple different `now` values.